### DISCUSSION TITLE
Integration with HSL does not work

### DISCUSSION THREAD

**Discussion by kutyel** [srid/haskell-template#93] [Q&A]
Hi! I just started a project with the template and used Nix Env Selector and VSCode + HLS and it is not able to detect the GHC version etc 😢

**Comment by srid** [comment]
> it is not able to detect the GHC version etc 

wdym

**Comment by kutyel** [comment]
The Glorious Glasgow Haskell Compilation System, version 9.4.4
cabal-install version 3.8.1.0
<img width="718" alt="Screenshot 2023-03-13 at 13 32 07" src="https://user-images.githubusercontent.com/5127501/224703009-ddf9a698-1fdf-45fd-ae45-9f32d47710b5.png">

**Reply by srid** [reply]
This means that your VSCode is not really in the nix shell's environment. You should have "Nix Env Selector" select the shell.nix environment. https://srid.ca/haskell-template/start

<img width="792" alt="image" src="https://user-images.githubusercontent.com/3998/224705412-b57bdf52-9115-4393-9fc2-80fa051c6d65.png">


If you run `nix develop -c haskell-language-server` on the command line you can confirm that haskell-template is correctly configured for HLS.

I personally no longer use "Nix Env Selector" and have switched to using nix-direnv along with https://marketplace.visualstudio.com/items?itemName=mkhl.direnv ... is the "Nix Env Selector" extension broken of late?

**Reply by kutyel** [reply]
Yes I think the Nix Env Selector extension does not work, how can I make HLS take the correct version of GHC with nix-direnv? 🤔

**Reply by srid** [reply]
I'll write a tutorial for this, but the short answer is:

1. install https://github.com/nix-community/nix-direnv
2. run `direnv allow` in your project directory
3. install https://marketplace.visualstudio.com/items?itemName=mkhl.direnv and activate the extension in VSCode (reloading it if asked)

**Reply by kutyel** [reply]
That worked, thanks!
